docs/design/coreclr/botr/type-system.md

```diff
@@ -193,19 +193,145 @@ This is enforced via an extensive and complicated set of enforcements within the
 - **ISSUE:** The signature walks performed are done with the normal signature walking code. This code is designed to load types as it walks the signature, but in this case the type load functionality is used with the assumption that no type load will actually be triggered.
 - **ISSUE:** Stackwalker requirements require support from not just the type system, but also the assembly loader. The Loader has had a number of issues meeting the needs of the type system here.
 
-Type System and NGEN
---------------------
+## Static variables
 
-The type system data structures are a core part of what is saved into NGEN images. Unfortunately, these data structures logically have pointers within them that point to other NGEN images. In order to handle this situation, the type system data structures implement a concept known as restoration.
-
-In restoration, when a type system data structure is first needed, the data structure is fixed up with correct pointers. This is tied into the type loading levels described in the [Type Loader](type-loader.md) Book of the Runtime chapter.
-
-There also exists the concept of pre-restored data structures. This means that the data structure is sufficiently correct at NGEN image load time (after intra-module pointer fixups and eager load type fixups), that the data structure may be used as is. This optimization requires that the NGEN image be "hard bound" to its dependent assemblies. See NGEN documentation for further details.
-
-Type System and Domain Neutral Loading
---------------------------------------
-
-The type system is a core part of the implementation of domain neutral loading. This is exposed to customers through the LoaderOptimization options available at AppDomain creation. Mscorlib is always loaded as domain neutral. The core requirement of this feature is that the type system data structures must not require pointers to domain specific state. Primarily this manifests itself in requirements around static fields and class constructors. In particular, whether or not a class constructor has been run is not a part of the core MethodTable data structure for this reason, and there is a mechanism for storing static data attached to the DomainFile data structure instead of the MethodTable data structure.
+Static variables in CoreCLR are handled by a combination of getting the "static base", and then adjusting it by an offset to get a pointer to the actual value.
+We define the statics base as either non-gc or gc for each field.
+Currently non-gc statics are any statics which are represented by primitive types (byte, sbyte, char, int, uint, long, ulong, float, double, pointers of various forms), and enums.
+GC statics are any statics which are represented by classes or by user defined structures.
+For user defined structure statics, the static variable is actually a pointer to a boxed instance of the structure.
```

Comment:
```suggestion
GC statics are any statics which are represented by classes or value-types.
For value-type statics, the static variable is actually a pointer to a boxed instance of the value-type.
```

I've not see anywhere we refer to value-types as "user defined structures". I'd prefer "value-types" or simple "structs".